apps/aens/src/aens.erl

```diff
@@ -85,11 +85,14 @@ get_name_hash(Name) when is_binary(Name) ->
 is_name(Bin) ->
     length(aens_utils:name_parts(Bin)) > 1.
 
+-spec name_to_name_hash(binary()) -> {ok, binary()} | {error, atom()}.
```

Comment:
Later on there is the expectation that this function returns either an ok tuple or an `{error, atom()}` one. In rare cases (names invalid according to IDNA) the function was returning a nested tuple instead. This is point 2 - the crash.